File: Algorithms/Sort/BubbleSort/bubbleSort.js

function bubbleSort( list, ascending=true )
{
    console.log( 'List Pre Sort:\n', list )

    for( let i = 0; i <= list.length - 1; i++ )
    {
        for( let j = 0; j <= list.length - 2; j++ )
        {
            num1 = list[ j ]
            num2 = list[ j + 1 ]

            if( ( num1 >= num2 && ascending ) || ( num1 <= num2 && !ascending ) )
            {
                list[ j + 1 ] = num1
                list[ j ] = num2
            }

        }
    }

    console.log( 'List Post Sort:\n', list )
}

list = []
for( let i = 0; i < 100; i++ ) list.push( Math.round( Math.random() * 100 ) )

// Set Bool to false for descending sort
bubbleSort( list, true )

File: Algorithms/Sort/BubbleSort/bubbleSort.py
import random
lenOfList = 100

randomIntList = [ random.randint( 0, 1000 ) for _ in range( lenOfList ) ]

def bubbleSort( List:list[int], ascending:bool ) :
    
    print( f'List Pre Sort:\n{List}')
    
    for n in range( len( List ) ) :
        for i in range( len( List ) - (n+1) ) :
            
            num1 = List[ i ]
            num2 = List[ i + 1 ]
                        
            if ( num1 > num2 and ascending ) or ( num1 < num2 and not ascending ):
                List[ i + 1 ] = num1
                List[ i ] = num2
        
    print( f'List Post Sort:\n{List}')
                        

# The list will be sorted according to the bool ascending
bubbleSort( randomIntList, ascending=True )